kolena/workflow/metrics/_plots.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import defaultdict
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Set
+
+from kolena._utils import log
+from kolena.workflow.evaluator import ConfusionMatrix
+from kolena.workflow.metrics._geometry import MulticlassInferenceMatches
+
+
+def compute_test_case_confusion_matrix(
+    all_matches: List[MulticlassInferenceMatches],
+    plot_title: str = "Confusion Matrix",
+) -> Optional[ConfusionMatrix]:
+    labels: Set[str] = set()
+
+    confusion_matrix: Dict[str, Dict[str, int]] = defaultdict(lambda: defaultdict(int))
+    for match in all_matches:
+        for gt, _ in match.matched:
+            actual_label = gt.label
+            confusion_matrix[actual_label][actual_label] += 1
+            labels.add(actual_label)
+
+        for gt, inf in match.unmatched_gt:
+            actual_label = gt.label
+            labels.add(actual_label)
+            if inf is not None:
+                predicted_label = inf.label
+                confusion_matrix[actual_label][predicted_label] += 1
+                labels.add(predicted_label)
+
+        labels.update(inf.label for inf in match.unmatched_inf)
```

Comment:
Do we need to add `unmatched_inf`'s label to `labels`?

Reply:
> conf matrices to be explicit
Can you elaborate more on this?

One thing I'm concerned about is that if there happens to be labels that were not part of any of the `matches` but are in `unmatched_inf`, [line 60](https://github.com/kolenaIO/kolena/pull/50/files/dd019b7598e23a70de0f5dea431d5fdadb5fe97e#diff-41991ae41fb36a2427881ebdff528b82d89b8daa07a6b1712d7945dcf6cd1144R60) will fail.

Reply:
I think the confusion matrix should be complete, containing all non-ignored labels, and should be a square (n by n). If a label was only in the `unmatched_inf`s, a user would gain more information from seeing a 0-column/row than having it absent from view. It might be misleading if `conf_mat(labels=[a,b,c])` and `conf_mat(labels=[a,b], ignore=[c])` produced the exact same matrix. (`ignore` is implicit from stratification / test case design)

`defaultdict` should handle line 60, but definitely good to add a test for that 👍